addons/web/static/tests/views/fields/daterange_field_tests.js

```diff
@@ -1240,4 +1240,56 @@ QUnit.module("Fields", (hooks) => {
             assert.containsOnce(target, ".fa-long-arrow-right");
         }
     );
+
+    QUnit.test("invalid empty date with optional end date",
+        async (assert) => {
+            serverData.models.partner.fields.date_end = { string: "Date End", type: "date" };
+            serverData.models.partner.records[0].date_end = "2017-02-08";
+            await makeView({
+                type: "form",
+                resModel: "partner",
+                serverData,
+                arch: `
+                    <form>
+                        <label for="date" string="Daterange" />
+                        <field name="date" widget="daterange" options="{'end_date_field': 'date_end'}" string="Planned Date" required="date or date_end"/>
```

Comment:
```suggestion
                        <field name="date" widget="daterange" options="{'end_date_field': 'date_end'}" string="Planned Date" required="date_end"/>
```
making date required if date is set doesn't make a lot of sense